lib/snowpack.ex

```diff
@@ -150,8 +152,19 @@ defmodule Snowpack do
   @spec query(conn, iodata, list, [query_option()]) ::
           {:ok, Snowpack.Result.t()} | {:error, Exception.t()}
   def query(conn, statement, params \\ [], options \\ []) when is_iodata(statement) do
-    case prepare_execute(conn, "", statement, params, options) do
+    # retry after 250 ms only once and do not rescue exceptions
+    retry with: Stream.take([250], 1), rescue_only: [] do
```

Comment:
is retrying only once enough? If we reach the limit of connections the failed connections will pile up, so this retry logic might only help a small number of operations 🤔 .

Reply:
although 250ms is enough time to solve some requests hehe

Reply:
We could retry more than once but I don't think it is needed in this case.

We are retrying because there is a connection in the connection pool that is bad and this query used that bad connection. When the query runs it will get the error denoting the bad connection and then signal the pool to disconnect and replace the connection.

This retry will get a new connection from the pool and use it. There is a chance that it will get assigned a connection that is also bad and in that case, it will fail, but that should be very rare with the new code that reactively recycles bad connections.

The 250ms is kinda arbitrary. That could probably be 50 or 100ms.